scanpy/readwrite.py

```diff
@@ -30,6 +31,12 @@
 """Available file formats for reading data. """
 
 
+class Empty(Enum):
+    token = 0
+
+_empty = Empty.token
```

Comment:
No and no. It’s only used if `None` doesn’t cut it, e.g.

- here: `compression=None` already means something, so we can’t use it because we’d risk to confuse people that it’s `'none'` somewhere and `None` somewhere else.
- in `slice`-like interfaces: `slice(1) == slice(None, 1, None)`, but `slice(1, None) == slice(1, None, None)`, so the default for the second argument needs to be `_empty`.

Reply:
Alright, I think that's what I meant by

> differentiate between "no option passed" and an explicit "no compression"

since `None` already has a meaning here. Another approach would be to do some meta programming and figure out whether an argument was passed. A sentinel value is probably better here.

Reply:
You’re right of course. I should have said “exactly, no, and no”, but ended up only replying to your last paragraph’s questions: It’s to differentiate, we shouldn’t adopt it elsewhere, and [this is the most pythonic way if we can’t use `None`](https://www.python.org/dev/peps/pep-0484/#support-for-singleton-types-in-unions).